### PR TITLE
Add missing gradient support

### LINKED ISSUE #1106: Slideshow: Content container should inherit gradient background
**Describe the current behavior**
With the image with text, if you have a gradient for the background and select that colour scheme it uses the gradient. However, for the slide show container it doesn't use the gradient. There doesn't seem to be any indication on when the gradient will be used, and when the plain colour will be used instead. 

**Describe the expected behavior**
Slideshow content container should inherit gradient background when a value is provided. 

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 2.5.0

image

### LINKED ISSUE #1265: Cards are not getting gradient support
**Describe the current behavior**

When adding a gradient, you'd expect cards to inherit that gradient but it's only applying the background 1 and 2 colors. 

**Describe the expected behavior**

When setting a gradient, it should take over.

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 2.5.0

**Possible solution**

Add the `gradient` class on the elements currently receiving the `color-scheme` classes.

**Additional context/screenshots**
[video](https://screenshot.click/22-01-3gmq4-rcurg.mp4)

### PR DESCRIPTION
**Pull request by kmeleta**
**Why are these changes introduced?**

Fixes #1265 
Fixes #1106 

Added gradient support to..
- Product cards, collection cards, and article cards
- Cart notification
- Search bar
- Search results popup
- Mobile menu
- Header subnav
- Slideshow content
- Image banner content
- Facet filter popups
- Product media modal
- Email signup banner
- Pickup availability drawer

Also.. 
- Added a solid background to quantity input to match other inputs ([see here](https://screenshot.click/02-26-n8vk2-v6v99.png))

**What approach did you take?**

I feel if text inputs and buttons aren't transparent (and thus may show a background gradient) then things like the quantity input and variant pills shouldn't be either. But we can certainly discuss and adjust as needed.

<details>
<summary>This looks right to me</summary>

</details>

<details>
<summary>This doesn't</summary>

</details>

**Other considerations**

**Testing steps/scenarios**
Test all sections noted above

**Demo links**

- [Editor](https://os2-demo.myshopify.com/admin/themes/127539249174/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by tyleralsbury** [comment]
An option for what you showed above could be for us to have each component - button, input, variant pill, etc... have a "use gradient" checkbox that enables or disabled the gradient class on them.  Then the merchant could decide component by component whether or not to have a gradient on them.  It would allow flexibility for different art directions.

A few concerns with that:

1. Contrast.  Contrast with a gradient is trickier to get right than a flat colour.  These are interactive elements which are especially important to have contrast well - though that's just generally a concern with any element that has a gradient.
2. We'd need some help text.  It would only apply if there is a gradient set for the colour scheme, so we'd need to indicate that for the checkbox.  The checkbox wouldn't do anything for many merchants if they aren't using gradients.

File: assets/base.css
Comment (kmeleta) [reply assets/base.css:1868]: Just the 1px border in the card settings.

File: assets/base.css
Comment (kmeleta) [reply assets/base.css:1872]: Good reminder. I forget about that issue because it varies based on how much radius is applied. Waiting for confirmation that my assumption is correct about the inputs having opaque backgrounds, but I'll fix the radius issue regardless.

File: assets/component-cart-notification.css
Comment (kmeleta) [reply assets/component-cart-notification.css:11]: Yeah.. that's most focus outlines right now. We just don't notice it until we apply a gradient bg. Not something I'll address here but I've discussed this with Meli in relation to another PR.

File: assets/base.css
Comment (sofiamatulis) [reply assets/base.css:1872]: Sounds good. I will re-review then 👍

**Review by melissaperreault — COMMENTED** [review]
Thank you Ken! 🎉 

What was reviewed:
- [x] Added gradient support to product cards, collection cards, and article cards
- [x] Added gradient support to cart notification
- [x] Added a solid background to quantity input to match other inputs
- [x] Other areas we missed the application
- [x] Played with global settings as well
- [ ] I couldn't check the product media modal, it was not opening on click. Can you?

### Observations:
- [ ] Search mode should inherit [the same header gradient](https://screenshot.click/22-02-ouox1-zbc6q.png) 
- [ ] Slideshow content container too, from that issue, if we can include 🙏 https://github.com/Shopify/dawn/issues/1106
- [ ] [The `pop-up block` modal ](https://screenshot.click/22-02-hfw7f-s3lr9.png)could support gradient 
- [ ] `Image banner` [content container should](https://screenshot.click/22-02-vhey9-01lqu.png) support gradient
- [ ] [Email sign up banner](https://screenshot.click/22-02-ap57c-8yff7.png) as well can support gradient.

While investigating the application of colors, I noticed that some dropdown and popups doesn't inherit the same colors. i.e. Currency and country select dropdown will get the `Background 1` while others dropdown `Background 2`. This is due to the color sheme at the bigger section level informing the solid fallback. Not sure what to do yet about that one but it was interesting to witness.
- [Search combo box from the header](https://screenshot.click/22-02-a5fdz-frzcv.png) doesn't inherits `Background 2` instead of `Background 1`, where as on the [Search template](https://screenshot.click/22-02-vu7ww-hec2d.png) it's `Background 1`. [Video reference](https://screenshot.click/22-02-10v2a-xkyau.mp4) of the inconsistency perceived. 
   - [ ] That being said, it could support gradient like `Cart notification` though, but open if we feel it is a better flow to keep the full solid colors for now, I'm ok with it.
- [ ] Drawers theme setting could benefit [of its own Color scheme setting](https://screenshot.click/22-02-ao8ol-1p7wt.png) so we can support gradient as well. _I think this fell between 2 chairs :)_  The application of the colors scheme don't make sense:
   - [ ] [Menu navigation](https://screenshot.click/22-02-1h554-6fdw5.png) should inherit `Background 1` and (Background 1 gradient) instead of `Background 2` solid, like Filter drawer [as you can see here](https://screenshot.click/22-02-4xax5-0nxv2.png).
   - [ ] [Filter drawer](https://screenshot.click/22-02-schh0-rgp4m.png) could also benefit from inheriting the `Background 1 gradient` when available, to align how we expect it for `Cart notification`.
   - [ ] Same for the Pickup availability drawer.
   - All of the above could be inherited by the same setting to have harmonious application.

File: assets/component-cart-notification.css
Comment (melissaperreault) [reply assets/component-cart-notification.css:14]: In the ideal world, the modal would support gradient as well. 🙏 I'll let you decide if this is a follow up to consider additional factors. Thanks!

**Comment by kmeleta** [comment]
Thanks @melissaperreault, I've added [what I believe to be] the correct colors and gradients to all the areas you mentioned with the exceptions discussed below

> The pop-up block modal could support gradient

This is proving difficult for some reason. I'd like to consider this separately and focus on getting the rest of these changes through.

> Drawers theme setting could benefit of its own Color scheme setting so we can support gradient as well. I think this fell between 2 chairs :)

For now I've just made all the popups/drawers mentioned inherit and apply the color/gradient from its closest parent. As in, header/nav drawers will use background 2 if header color scheme is set to background 2. Which I think makes sense. We've already dun' blown up the scope of this PR so let's deal with the new setting in another issue.

> Search combo box from the header doesn't inherits Background 2 instead of Background 1, where as on the Search template it's Background 1. Video reference of the inconsistency perceived.

I think it makes sense that the search results and header search bar could differ, given that the header search would be inheriting the header's color scheme. If the header color scheme was set to background 1 both search inputs would match. Let me know if we should discuss this one more.

> That being said, it could support gradient like Cart notification though, but open if we feel it is a better flow to keep the full solid colors for now, I'm ok with it.

I initially implemented this but I think we're better off keeping it solid and waiting until we know what we're doing with input gradients. It looks much better solid to me than with the [input solid and the dropdown gradient](https://screenshot.click/23-58-urc09-f5upu.png)

**Review by melissaperreault — COMMENTED** [review]
Thanks Ken! Looking real good! Went through the list and I have a few minor notes!  Should be good to ship after!

- [x] Product cards, collection cards, and article cards
- [x] Cart notification
- [x] Search bar
- [x] Search results popup
- [x] Mobile menu
   - [ ] [We can see the border picking](https://screenshot.click/22-02-s12vy-d8br9.png) the solid color, can we avoid and use the same as Pickup availability? Pickup availability drawer is a good reference.
- [x] Header subnav
- [x] Slideshow content
- [x] Image banner content
- [x] Facet filter popups
    - [ ] [1 layer deep doesn't support](https://screenshot.click/22-02-r9w7l-qu5xd.png) gradient yet
    - [ ] It's also causing the [header to overlap](https://screenshot.click/22-02-d6mku-8an3h.png)  
- [x] Product media modal
   - [ ] Noticed the border color source is not the same on the template and the modal. ([video reference](https://screenshot.click/22-02-bj4j8-4xt7e.mp4)) I'll open an issue if this is unrelated. Let me know!
- [x] Email signup banner
- [x] Pickup availability drawer

**Comment by kmeleta** [comment]
> Sorry, I forgot to review with keyboard! Also, sorry if I lost track about the direction we wanted to take to match how the focus is styled on menu links compared to other areas, can we not rely on that inner focus color finally?

No, we can't use gradients to do the box-shadow offset thing I'm afraid, so it will never be completely invisible. I think the best we can feasibly do in that regard is pick one of the colors that make up a gradient and use that. My goal in another PR (a follow up to this one even https://github.com/Shopify/dawn/pull/1337) is to remove the need for that color outline though which is why I'm not suggesting we go through the trouble of changing the color yet. In any case, it's too big of a change to do here. @melissaperreault

**Review by melissaperreault — DISMISSED** [review]
Works well! Thanks for all the care and details! I'll open the last issue for the remaining point for product modal media border! 🎉  🚢

File: assets/component-facets.css
Comment (sofiamatulis) [inline assets/component-facets.css:567]: This is still happening in the quantity input: https://screenshot.click/01-05-auu32-gr5ce.png (the element is not in focus), but the colors are not staying in the container

File: assets/component-facets.css
Comment (sofiamatulis) [inline assets/component-facets.css:702]: The product modal is behaving differently than the other modals and increasing in size when in focus. Is that expected? https://screenshot.click/01-08-r1phw-g5occ.mp4 

cc @melissaperreault

File: assets/component-facets.css
Comment (sofiamatulis) [inline assets/component-facets.css:749]: We have two outlines on focus here: grey and the red. Is that expected? 🤔 I dont see this on main

https://screenshot.click/01-09-nc3xl-htdlt.mp4

File: assets/component-facets.css
Comment (sofiamatulis) [inline assets/component-facets.css:791]: The share is overflowing here: https://screenshot.click/01-10-te46h-wg245.png I dont see this on main (https://screenshot.click/01-10-cb4ay-29ffm.png)

File: assets/component-list-menu.css
Comment (sofiamatulis) [inline assets/component-list-menu.css:11]: Not related to this PR but noticed this bug: https://screenshot.click/01-11-8oc29-vdpds.mp4

We are supposed to close one dropdown when another one opens, but if you start tabbing through it you are able to open the 2nd one without closing the first one

**Review by sofiamatulis — COMMENTED** [review]
Nice 👍 Just noted a few more things

File: assets/component-facets.css
Comment (kmeleta) [reply assets/component-facets.css:702]: You're seeing the focus outline "padding" as it were. Box-shadows don't allow gradients so when we try to impose the focus outline (via box-shadow) on a gradient background, the "invisible" background-colored padding is visible.

File: assets/component-facets.css
Comment (kmeleta) [reply assets/component-facets.css:700]: If it wasn't introduced here, let's open an issue so we don't add noise to this PR.

File: assets/component-facets.css
Comment (kmeleta) [reply assets/component-facets.css:567]: This probably should have been fixed separately, but the fix was easy enough. Should be resolved.

File: assets/component-list-menu.css
Comment (sofiamatulis) [reply assets/component-list-menu.css:11]: Created an issue here: https://github.com/Shopify/dawn/issues/1435

File: assets/component-facets.css
Comment (sofiamatulis) [reply assets/component-facets.css:700]: Created an issue: https://github.com/Shopify/dawn/issues/1436

Comment (melissaperreault) [reply assets/component-facets.css:700]: Good catch Sofia! Thanks for opening the issue!

**Review by tyleralsbury — APPROVED** [review]
Tested + Code reviewed.  The demo store is very flashy with all the gradients. 🤣 

I'd love to see an actual store that wacky - maybe a candy shop!

**Pull request merged**
